readthedocs/core/static-src/core/js/doc-embed/footer.js

```diff
@@ -78,6 +79,10 @@ function init() {
             }
             injectFooter(data);
             setupBookmarkCSRFToken();
+
+            if (!data.ad_free_user && !data.ad_free_project) {
+                sponsorship.init();
```

Comment:
I imagine this is going to add a decent bit of latency to ad viewing, I wonder if we should do the ads request always, but vary the display based on this data?

Reply:
It will add some latency, but I'm not sure how you can actually do what you're saying. Either you add the `ethicalads.js` meaning you want an ad or you don't. To determine that, you need to know some additional data about the user or project. Having a stripped down API that just gets the project/user data would likely be faster although there would be an additional concurrent API call.

Reply:
I think we do the ad request, but we can hide it with CSS until we confirm we want to show it. That way as soon as the footer responds, we can display the ad, instead of doing another full round trip.

Reply:
I don't think we should be hiding ads with CSS and showing them based on API responses. I think this will lead to complication. Instead, perhaps there's a way to add some integration points to the ad client so we can control when requests for an ad are made.

Reply:
I think we can start with this approach for now. But I do think trying to reduce latency on the ad display is important.

Reply:
> I don't think we should be hiding ads with CSS and showing them based on API responses

Is it possible to do the request for the ad, save all the data needed (view URL, click URL, image URL, text, etc) without creating the HTML element to display it yet, until we receive the response from the footer and _there_ create the HTML element and show it?

I assume the ad-client is not thought to work like that at this point, but I think we could have the best of both ideas with lower latency.

Reply:
Yea, this is what we settled on 👍